vassal-app/src/main/java/VASSAL/counters/BoundsTracker.java

```diff
@@ -29,25 +31,27 @@
  * added pieces belonged.
  */
 public class BoundsTracker {
-  private final Set<Map> maps;
+  private final List<Pair<Rectangle, Map>> regions;
 
   public BoundsTracker() {
-    maps = new HashSet<>();
+    regions = new LinkedList<>();
```

Comment:
Or possibly still a `HashSet`, because that gets you deduplication.

Reply:
Or, maybe a `HashMap<Map, ArrayList<Rectangle>>`? Depends on just how many elements there are typically. If these things can be huge, this will be more efficient. If small, then the array of pairs is likely better.

Reply:
I think using a HashMap is not necessary for performance reasons. Might be nice to see your regions separated by map ... but it does make the code more complex.

Reply:
You're probably right that these lists tend to be small.